ui/ducks/metamask/metamask.js

```diff
@@ -162,6 +163,12 @@ export default function reduceMetamask(state = initialState, action) {
         participateInMetaMetrics: action.value,
       };
 
+    case actionConstants.SET_DATA_COLLECTION_FOR_MARKETING:
```

Comment:
What is this for? Generally we shouldn't adding anything to this Redux slice, it's meant as a mirror of the background state. The actions here other than the background state update action are all examples of technical debt that we have not cleaned up yet.

Reply:
This is to mirror the implementation of SET_PARTICIPATE_IN_METAMETRICS. We are adding the marketing toggle flag to the state as well (dataCollectionForMarketing)

Reply:
This seems like a misunderstanding. `SET_PARTICIPATE_IN_METAMETRICS` shouldn't be here either

Reply:
These are good points, and these changes will be postponed for a later date given our time constraints. Both SET_PARTICIPATE_IN_METAMETRICS and SET_DATA_COLLECTION_FOR_MARKETING should be deleted from the redux state.